tests/unittests/MLTest.cpp

```diff
@@ -199,6 +199,78 @@ TEST_P(MLTest, learnXor) {
     EXPECT_NEAR(resH.at({i, 0}), (a ^ b), 0.1);
   }
 }
+/// Learn the logarithmic function.
+TEST_P(MLTest, learnLog) {
+  unsigned numInputs = 50;
+  unsigned batchSize = 5;
+  EE_.getConfig().learningRate = 0.07;
+  EE_.getConfig().batchSize = batchSize;
+
+  auto &mod = EE_.getModule();
+  Function *F = mod.createFunction("learnLog");
+
+  auto *A =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "A",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+  auto *Ex =
+      mod.createVariable(ElemKind::FloatTy, {batchSize, 1}, "Ex",
+                         VisibilityKind::Public, Variable::TrainKind::None);
+
+  Node *O = F->createFullyConnected("fc1", A, 4);
+  O = F->createTanh("tanh1", O);
+  O = F->createFullyConnected("fc2", O, 1);
+  O = F->createRegression("reg", O, Ex);
+  auto *result = F->createSave("ret", O);
+
+  // Set the training data.
+  Tensor trainingSet(ElemKind::FloatTy, {numInputs, 1});
+  Tensor trainingLabels(ElemKind::FloatTy, {numInputs, 1});
+
+  auto TS = trainingSet.getHandle<>();
+  auto TL = trainingLabels.getHandle<>();
+
+  // Set the training date as floating number from range [0.75, 1.5).
+
+  const float LO = 0.75; // Lower bound of training data.
+  const float HI = 1.5;  // Upper bound of training data.
+  for (size_t i = 0; i < numInputs; i++) {
+    // Generate a floating number in the range of [LO,HI).
+    float a = LO + i * (HI - LO) / numInputs;
+    TS.at({i, 0}) = a;
+    TL.at({i, 0}) = std::log(a);
+  }
+
+  Function *TF = glow::differentiate(F, EE_.getConfig());
+  EE_.compile(CompilationMode::Train, TF);
+
+  // Train the network:
+  EE_.runBatch(1000, {A, Ex}, {&trainingSet, &trainingLabels});
```

Comment:
Do we really need a 1000 iterations? Can we reduce this to a smaller number?

Reply:
the smallest number i tried that can pass the test, is 800. It seems difficult to reduce it to even smaller.  

I might have to modify other parameters so that each batch may learn faster, if we really want to reduce batch number. However the learning rate is relatively high now(0.07), there does not seem to have much space for radical changes. (Please correct me if im wrong)

Also the run time of 800 batches and 1000 batches do not have much differences (~100ms in total). Maybe we just stick to this batch number?